go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/azure/azure-dev
 
-go 1.21
+go 1.23
```

Comment:
Do this will bring in fixes to the standard library with how different types of reparse points are handled on windows, which I think will fix the dev drive issues we saw, but I will need to confirm on a windows machine.  They are gated behind a compat flag that is on for go 1.23 and higher.